### ISSUE TITLE
A cluster node never reaches "Up" status if observed as unreachable by another nodes 

### ISSUE THREAD

**Issue by jdevelop** [akka/akka#19713]
I have some test cluster, that runs on **Akka 2.4.2-RC2**. At some point under severe network load - some node might be reported as "Unreachable" by other nodes. If this "unreachable" node is restarted - then it gets into the "Joining" status and never reaches "Up" status. It is still reported as "Unreachable" in the cluster status. I have shut down the node for hours - but it didn't help, it remains in the "unreachable" list until I restart some nodes from the list of nodes that have reported the node as "unreachable".

Some examples of cluster status from https://groups.google.com/forum/#!topic/akka-user/UT8jIh_EfV0

```
    {
      "address": "akka.tcp://HttpCluster@192.168.0.205:2551",

      "status": "Up",
      "roles": [
        
      ]
    },
```

```
"unreachable": [
    {
      "node": "akka.tcp://HttpCluster@192.168.0.205:2551",
      "observed-by": [
        "akka.tcp://HttpCluster@10.9.0.10:2551",
        "akka.tcp://HttpCluster@10.9.0.11:2551",
        "akka.tcp://HttpCluster@192.168.0.200:2551",
        "akka.tcp://HttpCluster@192.168.0.202:2551",
        "akka.tcp://HttpCluster@192.168.0.203:2551"
      ]
    }
  ]
```

The same behavior is observed with **2.4-SNAPSHOT**

**Comment by patriknw** [comment]
We need instructions of how to reproduce the problem. I have never seen this issue.

**Comment by hveiga** [comment]
Hi @patriknw, I am seeing this issue as well in our cluster. I am trying to come up with a test to reproduce it.

**Comment by jdevelop** [comment]
@patriknw I'm trying to get the test case, but couldn't figure out the exact steps. What I know is that sometimes a random node is noticed as "unreachable" (may be a GC pause or a network error) and then it just sits there. Restarting that unreachable node doesn't help, the only thing that helps - restart the nodes that "observe" the unreachable one and **then** restart the unreachable one.

Any way to retrieve some debug information/cluster state that might help you to nail down the problem?

**Comment by hveiga** [comment]
I have put together these two tests, with my different configurations where one succeeds and the other one fails. Honestly, I am not sure if my configuration can be part of the issue:

Working test: https://gist.github.com/hveiga/3c562a6da915935e3ba5
Not working test: https://gist.github.com/hveiga/77f98c6d0070150e76c6

**Comment by patriknw** [comment]
@hveiga Thanks for trying to reproduce. The second version fails on barrier timeout, so adding this makes it work for me:

    akka.testconductor.barrier-timeout = 150s

That is indeed a weird configuration and I hope it was only an attempt to provoke things and not something you intend to use in production.

**Comment by patriknw** [comment]
@jdevelop Logs would help me understand what is going on. Logs on INFO level are normally enough, but DEBUG level might be needed. Your configuration is also of interest.

**Comment by hveiga** [comment]
Thanks @patriknw . I will keep trying to see if I can reproduce it.

**Comment by hveiga** [comment]
@patriknw The issue I am seeing is actually different than this, should I open a thread in the akka user group or a ticket here? My issue does not involve any shutdown or restart. I am still working on a test to reproduce it.

**Comment by patriknw** [comment]
Ok, if you think it is an issue then create an issue, if question or uncertain start with akka-user. thanks

**Comment by jdevelop** [comment]
I added some logging on the cluster, and was able to reproduce this situation here (sorry, still no test case - that only happens in "live" cluster, I can not isolate problem)

the master (seed) node address is 192.168.0.200, the address of joining-forever node - 10.9.0.12


### 10.9.0.12 logs, timezone: CET
```
INFO   | jvm 1    | 2016/02/17 04:03:02 | 2016-02-17 04:03:02,137 INFO   [Remoting] Starting remoting 
INFO   | jvm 1    | 2016/02/17 04:03:02 | 2016-02-17 04:03:02,538 INFO   [Remoting] Remoting started; listening on addresses :[akka.tcp://HttpCluster@10.9.0.12:2551] 
INFO   | jvm 1    | 2016/02/17 04:03:02 | 2016-02-17 04:03:02,542 INFO   [Remoting] Remoting now listens on addresses: [akka.tcp://HttpCluster@10.9.0.12:2551] 
INFO   | jvm 1    | 2016/02/17 04:03:02 | 2016-02-17 04:03:02,568 INFO   [Cluster(akka://HttpCluster)] Cluster Node [akka.tcp://HttpCluster@10.9.0.12:2551] - Starting up... 
INFO   | jvm 1    | 2016/02/17 04:03:02 | 2016-02-17 04:03:02,660 INFO   [Cluster(akka://HttpCluster)] Cluster Node [akka.tcp://HttpCluster@10.9.0.12:2551] - Registered cluster JM
X MBean [akka:type=Cluster]                                                                                                                                                        INFO   | jvm 1    | 2016/02/17 04:03:02 | 2016-02-17 04:03:02,660 INFO   [Cluster(akka://HttpCluster)] Cluster Node [akka.tcp://HttpCluster@10.9.0.12:2551] - Started up successful
ly                                                                                                                                                                                 INFO   | jvm 1    | 2016/02/17 04:03:02 | 2016-02-17 04:03:02,689 INFO   [Cluster(akka://HttpCluster)] Cluster Node [akka.tcp://HttpCluster@10.9.0.12:2551] - Metrics will be retre
ived from MBeans, and may be incorrect on some platforms. To increase metric accuracy add the 'sigar.jar' to the classpath and the appropriate platform-specific native libary to 'java.library.path'. Reason: java.lang.ClassNotFoundException: org.hyperic.sigar.Sigar                                                                                              INFO   | jvm 1    | 2016/02/17 04:03:02 | 2016-02-17 04:03:02,705 INFO   [Cluster(akka://HttpCluster)] Cluster Node [akka.tcp://HttpCluster@10.9.0.12:2551] - Metrics collection ha
s started successfully                                                                                                                                                             INFO   | jvm 1    | 2016/02/17 04:03:03 | 2016-02-17 04:03:03,849 INFO   [Cluster(akka://HttpCluster)] Cluster Node [akka.tcp://HttpCluster@10.9.0.12:2551] - Welcome from [akka.tc
p://HttpCluster@192.168.0.200:2551]
```

### 192.168.0.200 logs, timezone EST
```
INFO   | jvm 1    | 2016/02/16 01:11:15 | 2016-02-16 01:11:15,596 WARN   [Remoting] Association to [akka.tcp://HttpCluster@10.9.0.12:2551] having UID [-1885373876] is irrecoverably failed. UID is now quarantined and all messages to this UID will be delivered to dead letters. Remote actorsystem must be restarted to recover from this situation. 
INFO   | jvm 1    | 2016/02/16 22:03:03 | 2016-02-16 22:03:03,939 INFO   [Cluster(akka://HttpCluster)] Cluster Node [akka.tcp://HttpCluster@192.168.0.200:2551] - Node [akka.tcp://HttpCluster@10.9.0.12:2551] is JOINING, roles []
```

The timezones there are different, but the clocks are in sync (ntp is up).

There was something weird with the node 10.9.0.12 - that is it has failed irrecoverably for some reason, but after the restart it gets "stuck"

Is there anything (besides getting the test case, which I'm trying to accomplish at the time) I could tweak to nail down the problem? Debug logs would probably cause some huuuge logs, perhaps there could be some interesting packages to limit the logging for?

**Comment by patriknw** [comment]
Yeah, more logs, to see more context around the issue would be good. Here are some things you can turn on: http://doc.akka.io/docs/akka/2.4.2/scala/logging.html#Auxiliary_remote_logging_options

**Comment by jdevelop** [comment]
Is there a way to log only _service_ messages sent by the cluster, no the custom ones that were sent by my app?

**Comment by patriknw** [comment]
well, you can use slf4j/logback and define logging filters, or you can filter the logs after the fact with `grep`

**Comment by patriknw** [comment]
I think we have fixed this

**Comment by jdevelop** [comment]
@patriknw is there a corresponding PR or a commit for this?

**Comment by patriknw** [comment]
The issue I was thinking about: https://github.com/akka/akka/issues/19274
Fixed in 2.4.2

**Comment by patriknw** [comment]
The issue I was thinking about: https://github.com/akka/akka/issues/19274
Fixed in 2.4.2
